Setup.hs

```diff
@@ -29,67 +25,78 @@ linuxBuild :: Bool -> WithEnvAndVerbosity
 linuxBuild True env verbosity = do
     noticeNoWrap verbosity "Static linking."
     -- the target-feature=-crt-static is needed so that C symbols are not included in the generated rust libraries. For more information check https://rust-lang.github.io/rfcs/1721-crt-static.html
-    rawSystemExitWithEnv
-        verbosity
-        "cargo"
-        (addFeatures ["build", "--release", "--manifest-path", "rust-src/Cargo.toml", "--target", "x86_64-unknown-linux-musl"])
-        (("RUSTFLAGS", "-C target-feature=-crt-static") : env)
-    let copyLib lib = do
-            let source = "../rust-src/target/x86_64-unknown-linux-musl/release/lib" ++ lib ++ ".a"
-                target = "./lib/lib" ++ lib ++ ".a"
+    let makeLib lib = do
+            let libName = fst lib
+                libFeatures = snd lib
+            -- the target-feature=-crt-static is needed so that C symbols are not included in the generated rust libraries. For more information check https://rust-lang.github.io/rfcs/1721-crt-static.html
+            rawSystemExitWithEnv
+                verbosity
+                "cargo"
+                (["rustc", "--release", "--manifest-path", "rust-src/" ++ libName ++ "/Cargo.toml", "--target", "x86_64-unknown-linux-musl", "--crate-type", "staticlib"] ++ libFeatures)
+                (("RUSTFLAGS", "-C target-feature=-crt-static") : env)
+            let source = "../rust-src/target/x86_64-unknown-linux-musl/release/lib" ++ libName ++ ".a"
+                target = "./lib/lib" ++ libName ++ ".a"
             rawSystemExit verbosity "ln" ["-s", "-f", source, target]
             noticeNoWrap verbosity $ "Linked: " ++ target ++ " -> " ++ source
-    mapM_ copyLib concordiumLibs
+    mapM_ makeLib concordiumLibs
 linuxBuild False env verbosity = do
     noticeNoWrap verbosity "Dynamic linking."
-    rawSystemExitWithEnv verbosity "cargo" (addFeatures ["build", "--release", "--manifest-path", "rust-src/Cargo.toml"]) env
-    let copyLib lib = do
-            let source = "../rust-src/target/release/lib" ++ lib
-                target = "./lib/lib" ++ lib
+    let makeLib lib = do
+            let libName = fst lib
+                libFeatures = snd lib
+            rawSystemExitWithEnv verbosity "cargo" (["rustc", "--release", "--manifest-path", "rust-src/" ++ libName ++ "/Cargo.toml", "--crate-type", "cdylib"] ++ libFeatures) env
+            notice verbosity "Linking libraries to ./lib"
```

Comment:
I think here (and in the other places like this) it would be better to log "Linking library <libname> to ./lib", since this occurs for each library.